Problem when trying to logon to dedicated server

<!-- If you are making a bug report, please fill out this template as best as possible, this will help 
     us to understand your issue much easier. -->
### Minecolonies version
minecolonies-1.12.2-0.10.671-ALPHA-universal
### Expected behavior
To work when i log on to server
### Actual behaviour
[22:19:51] [Server thread/ERROR] [net.minecraft.server.MinecraftServer]: Encountered an unexpected exception
net.minecraft.util.ReportedException: Ticking entity
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:773) ~[MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) ~[nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_241]
Caused by: java.lang.ClassCastException: net.minecraft.pathfinding.PathPoint cannot be cast to com.minecolonies.coremod.entity.pathfinding.PathPointExtended
	at com.minecolonies.coremod.entity.pathfinding.MinecoloniesAdvancedPathNavigate.func_75508_h(MinecoloniesAdvancedPathNavigate.java:436) ~[MinecoloniesAdvancedPathNavigate.class:?]
	at net.minecraft.pathfinding.PathNavigate.func_75501_e(SourceFile:188) ~[ze.class:?]
	at com.minecolonies.coremod.entity.pathfinding.MinecoloniesAdvancedPathNavigate.func_75501_e(MinecoloniesAdvancedPathNavigate.java:128) ~[MinecoloniesAdvancedPathNavigate.class:?]
	at net.minecraft.entity.EntityLiving.func_70626_be(EntityLiving.java:766) ~[vq.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.java:2359) ~[vp.class:?]
	at net.minecraft.entity.EntityLiving.func_70636_d(EntityLiving.java:577) ~[vq.class:?]
	at net.minecraft.entity.EntityAgeable.func_70636_d(EntityAgeable.java:178) ~[vd.class:?]
	at com.minecolonies.coremod.entity.citizen.EntityCitizen.func_70636_d(EntityCitizen.java:1058) ~[EntityCitizen.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2179) ~[vp.class:?]
	at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295) ~[vq.class:?]
	at net.minecraft.world.World.func_72866_a(World.java:1996) ~[amu.class:?]
	at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:832) ~[oo.class:?]
	at net.minecraft.world.World.func_72870_g(World.java:1958) ~[amu.class:?]
	at net.minecraft.world.World.func_72939_s(World.java:1762) ~[amu.class:?]
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:613) ~[oo.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:767) ~[MinecraftServer.class:?]

<!-- If you have any videos, images or logs relating to this issue, please post them here also. -->

### Steps to reproduce the problem
just try to logon and crash

--- a/src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java
+++ b/src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java
@@ -427,7 +427,7 @@ public class MinecoloniesAdvancedPathNavigate extends AbstractAdvancedPathNaviga
         final int curNodeNext = curNode + 1;
         if (curNodeNext < currentPath.getCurrentPathLength())
         {
-            if (!(currentPath.getPathPointFromIndex(curNode) instanceof PathPointExtended))
+            if (!(currentPath.getPathPointFromIndex(curNode) instanceof PathPointExtended) || !(currentPath.getPathPointFromIndex(curNodeNext) instanceof PathPointExtended))
             {
                 currentPath = convertPath(currentPath);
             }

